# Promise.all

The Promise.all() static method `takes an iterable of promises as input` and `returns a single Promise`. All promises will `run parallelly`

This returned promise `fulfills` when `all of the input's promises fulfill` and `rejects` when `any of the input's promises rejects`, with this first rejection reason

```
  const promisify = (item, delay) => {
    return new Promise((resolve, reject) => {
      setTimeout(() => {
        resolve(item);
      }, delay);
    });
  };

  const shortDelay = () => Promise.resolve("short");
  const longDelay = () => promisify("long", 3000);

  const runParallel = async () => {
    console.log("running...");
    const startTime = new Date().getSeconds();

    const result = await Promise.all([shortDelay(), longDelay()]);

    const endTime = new Date().getSeconds();
    console.log(`done in ${endTime - startTime} second`, result);
  };

  runParallel();
```

`Throw error` if `any promise rejects`

```
  const runParallel = async () => {
    const result = await Promise.all([
      Promise.resolve("short"),
      Promise.reject("There is any error"),
    ]);

    console.log(`done`, result);
  };

  runParallel();
```
